lib/chef/resource/windows_firewall_rule.rb

```diff
@@ -195,12 +228,53 @@ def firewall_command(cmdlet_type)
 
           cmd
         end
+
+        def define_resource_requirements
```

Comment:
Traditionally we stick this above the actions.

Reply:
I would agree with that, but as I have tested this, it's not possible to move it out of the `action_class` block.
So it is in the same block like we have right now in the `locale` resource (https://github.com/chef/chef/blob/master/lib/chef/resource/locale.rb#L81)